api/app/alert.py

```diff
@@ -30,6 +35,12 @@ def _pteam_tag_page_link(pteam_id: UUID | str, tag_id: UUID | str) -> str:
     )
 
 
+def _pteam_service_tab_link(pteam_id: UUID | str, service_id: UUID | str) -> str:
+    baseurl = os.getenv("WEBUI_URL", "http://localhost")
+    baseurl += "" if baseurl.endswith("/") else "/"
+    return urljoin(baseurl, f"?pteamId={pteam_id}&serviceId={service_id}")
```

Comment:
queryパラメータの組み立てには、 urlencodeを使う方が良さそうです (自前実装だと適切にURLエンコードされない)
https://docs.python.org/ja/3/library/urllib.parse.html#urllib.parse.urlencode

(他の自前組み立てしている部分は別途修正)

```suggestion
    baseurl = os.getenv("WEBUI_URL", "http://localhost")
    baseurl += "" if baseurl.endswith("/") else "/"
    params = urlencode({"pteamId": pteam_id, "serviceId": service_id})
    return urljoin(baseurl, f"?{params}")
```

またurllib のサンプルだと、 query parameter はURLと文字列連結にしているのでそれでもいいかもしれません
https://docs.python.org/ja/3/library/urllib.request.html#urllib-examples

Reply:
末尾の`/` を必須にしている理由はありますか？ 必要性の判断ができないのでPRのコメントなどで補足お願いします